### PR TITLE
FIX Warning OF pytorch-0.3.0: Implicit dimension choice for softmax

### PR DESCRIPTION
**Pull request by ScarletPan**
It has been deprecated. Change the call to include dim=X as an argument.

### CODE REVIEW

**Comment by srush** [comment]
Yeah, we need to fix this. You can leave it open

**Comment by ScarletPan** [comment]
The same in ModelConstructor.py 189; ConvMultiStepAttention.py 71; MultiHeadedAttn 68; I think you guys can fix these minor warnings once you're ready to switch to 0.3.0 : )

**Comment by srush** [comment]
Will do.
